entity/index.js

```diff
@@ -562,11 +562,12 @@ EntityGenerator.prototype.askForFields = function askForFields() {
                 fieldValidateRulesMinlength: props.fieldValidateRulesMinlength,
                 fieldValidateRulesMaxlength: props.fieldValidateRulesMaxlength,
                 fieldValidateRulesPattern: props.fieldValidateRulesPattern,
+                fieldValidateRulesPatternJava: props.fieldValidateRulesPattern.replace(/\\/g, '\\\\'),
```

Comment:
props.fieldValidateRulesPattern is not always defined so this line result that no other validation rule works without the regex one.

see stacktrace
```bash
$ jhe Oke
The entity Oke is being created.
Generating field #1
? Do you want to add a field to your entity? Yes
? What is the name of your field? lala
? What is the type of your field? String
? Do you want to add validation rules to your field? Yes
? Which validation rules do you want to add? Maximum length
? What is the maximum length of your field? 100
/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:579
    throw e;
          ^
TypeError: Cannot read property 'replace' of undefined
    at EntityGenerator.<anonymous> (/Users/anders/Workspace/generator-jhipster/entity/index.js:565:79)
    at EntityGenerator.<anonymous> (/Users/anders/Workspace/generator-jhipster/node_modules/yeoman-generator/lib/base.js:233:7)
    at PromptUI.onCompletion (/usr/local/lib/node_modules/yo/node_modules/inquirer/lib/ui/prompt.js:69:10)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.completed (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:1977:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:1914:14)
    at AnonymousObserver.tryCatcher (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:567:29)
    at AutoDetachObserverPrototype.completed (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:5032:56)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:1914:14)
    at InnerObserver.onCompleted (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:3345:65)
    at InnerObserver.tryCatcher (/usr/local/lib/node_modules/yo/node_modules/inquirer/node_modules/rx/dist/rx.js:567:29)
```